src/OpenFeature/Api.cs

```diff
@@ -248,18 +223,17 @@ public EvaluationContext GetContext()
         /// Once shut down is complete, API is reset and ready to use again.
         /// </para>
         /// </summary>
-        public async Task Shutdown()
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to cancel any async side effects.</param>
+        public async Task ShutdownAsync(CancellationToken cancellationToken = default)

```

Comment:
Ya, I think this is an important discussion. I opened this PR to try to get to the bottom of this. See conversation on slack: https://cloud-native.slack.com/archives/C06EDTWU25B/p1714590325481829?thread_ts=1714590325.481829&cid=C06EDTWU25B

Reply:
I'd be happy to remove cancellation tokens from `SetProviderAsync` and `ShutdownAsync` entirely, in which case it would be better to implement `IDisposableAsync`.

Reply:
Let's wait to see what others say, but it seems to me that we can just remove CTs from init/shutdown, and go back to `IDisposable`. Let me know if you disagree.

Reply:
I've removed cancellation from `ShutdownAsync`